src/transformers/models/whisper/convert_openai_to_hf.py

```diff
@@ -51,6 +78,49 @@
 }
 
 
+def _alignment_head_indices(
+    alignment_heads: bytes,
+    n_text_layer: int,
+    n_text_head: int,
+) -> list[tuple[int, int]]:
+    arr = np.frombuffer(gzip.decompress(base64.b85decode(alignment_heads)), dtype=bool)
+    mask = arr.reshape(n_text_layer, n_text_head)
+    return [(int(i), int(j)) for i, j in zip(*np.where(mask))]
+
+
+def _get_generation_config(
+    n_text_layer: int,
+    n_text_head: int,
+    is_multilingual: bool,
+    num_languages: int = 100,
+    openai_version: Optional[str] = None,
+) -> GenerationConfig:
+    """
+    Loads the appropriate generation config from HF repo
+    """
+    if not is_multilingual:
+        repo = "openai/whisper-medium.en"
+    elif num_languages < 100:
+        repo = "openai/whisper-large-v2"
+    else:
+        repo = "openai/whisper-large-v3"
+
+    gen_cfg = GenerationConfig.from_pretrained(repo)
+    if openai_version is not None:
+        gen_cfg.alignment_heads = _alignment_head_indices(_ALIGNMENT_HEADS[openai_version], n_text_layer, n_text_head)
```

Comment:
I actually don't think we need to do this at all. All the pre-trained checkpoints already have the alignment heads in the generation config already, e.g. for large-v2: https://huggingface.co/openai/whisper-large-v2/blob/696465c62215e36a9ab3f9b7672fe7749f1a1df5/generation_config.json#L2

That means the attribute `.alignment_heads` is **already** correct and present for these checkpoints! So no need to load them from base64 encoding and override the attribute, since it'll result in no NET change to the alignment heads. TLDR: let's just keep them as they are `from_pretrained`!

We can remove the attribute for custom checkpoints as you have below